### PR TITLE
Save the main blocks workspace as a PNG

### PR DESCRIPTION
**Pull request by shrutirij**
Try it - http://ai2-experiments.appspot.com/

Open blocks view, right-click on the editor - an option "Download Blocks as Image" should appear.

Design doc [here](https://docs.google.com/document/d/1-CkBJsnl8Gcw_EvZqR-wavDyC4DpN9xslxxRWJKiHhg/edit).

Known bugs (help needed):
1. Erratic behavior in Safari
2. Files with large number of blocks causes out-of-memory error in Chrome
3. Downloaded PNG has surrounding whitespace (because the Blockly editor automatically expands) - cropping would be nice, but cannot figure out an efficient way to do so.

CC: @josmas

### CODE REVIEW

**Comment by jisqyv** [comment]
@shrutirij You included some copyrighted code. Is it licensed with the Apache 2.0 license or compatible?

File: appinventor/appengine/war/blocklyframe.html [hunk 1]
```diff
@@ -10,6 +11,7 @@
     <style>
       html, body {
       background-color: #fff;
+      height: 100%;
```
Comment (josmas) [inline appinventor/appengine/war/blocklyframe.html]: what's the impact of this change?

File: appinventor/blocklyeditor/src/warningIndicator.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ Blockly.WarningIndicator.prototype.top_ = 0;
 Blockly.WarningIndicator.prototype.createDom = function() {
 
   this.svgGroup_ = Blockly.createSvgElement('g',
-      {}, null);
+      {'id': "indicatorWarning"}, null);
```
Comment (josmas) [inline appinventor/blocklyeditor/src/warningIndicator.js]: is this necessary? or left from a previous modification?

**Comment by josmas** [comment]
I don't think the code is licensed; if I recall correctly it is copied from a blogpost, but it should include a link to that blogpost. @jisqyv @shrutirij

File: appinventor/appengine/war/blocklyframe.html (hunk 1 above)
Comment (shrutirij) [inline appinventor/appengine/war/blocklyframe.html]: When I change the doctype, the blocks editor does not fill its parent element completely [quirks mode vs standard mode]. This change ensures that the blocks editor fills the whole height.

File: appinventor/blocklyeditor/src/warningIndicator.js (hunk 2 above)
Comment (shrutirij) [inline appinventor/blocklyeditor/src/warningIndicator.js]: This is used to identify and hide the warning indicator when creating the PNG.

**Comment by shrutirij** [comment]
@jisqyv  @josmas 
I added that copyright line. What is the appropriate way to do this?

The code is MIT licensed and is from [here](https://github.com/exupero/saveSvgAsPng).

File: appinventor/blocklyeditor/src/exportBlocksImage.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+goog.provide('Blockly.ExportBlocksImage');
+
+goog.require('goog.Timer');
+
+
+/**
+ * saveSvgAsPng
+ * Copyright (c) 2014 Eric Shull
+ */
```
Comment (jisqyv) [inline appinventor/blocklyeditor/src/exportBlocksImage.js]: Add a line here stating that it is released under the MIT License

**Comment by halatmit** [comment]
@josmas @jisqyv Is this ready to be reviewed and merged?

**Comment by halatmit** [comment]
@afmckinney Andrew, please review.

**Comment by afmckinney** [comment]
LGTM, On gerrit and merged.

**Pull request closed without merging**